sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -143,25 +143,25 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
 
     // For CREATE TABLE [AS SELECT], we should use the v1 command if the catalog is resolved to the
     // session catalog and the table provider is not v2.
-    case c @ CreateTableStatement(
-         SessionCatalogAndTable(catalog, tbl), _, _, _, _, _, _, _, _, _, _, _) =>
+    case c @ CreateV2Table(ResolvedDBObjectName(catalog, name), _, _, _, _, _) =>
```

Comment:
I think we should only change the plan if the catalog is session catalog. The `CreateTableExec` should be responsible to convert `TableSpec` to table properties.